v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -29,392 +29,585 @@ Note: More types will be added as needed.
     uint8_t
 
 #define DECLARE_BASIC_TYPE(type_name) \
-    extern const LOG_CONTEXT_PROPERTY_TYPE_IF MU_C2(type_name, _log_context_property_type);
+    int LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)(void* dst_value, type_name src_value); \
+    size_t LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(type_name)(type_name src_value); \
+    extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(type_name); \
 
 MU_FOR_EACH_1(DECLARE_BASIC_TYPE, SUPPORTED_BASIC_TYPES)
 ```
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
```

Comment:
>void* [](http://example.com/codeflow?start=51&length=5)

I would argue that this is const void*, as the function doesn't take ownership of the "property_value", it just looks what's inside.

Like... 

```
const int64_t u=89;
char buffer[100];
LOG_CONTEXT_PROPERTY_TYPE_TO_STRING(&u, buffer, sizeof(buffer))
```

might throw compiler warnings.

Reply:
It's absolutely const void*, so changed